workflow/lib/shared/illumination_correction.py

```diff
@@ -145,3 +139,163 @@ def calculate_ic_field(
+        if radius <= 10:
+            shrink_factor = 1
+            trim = 0.12  # Trim 24% in x and y
+        elif radius <= 30:
+            shrink_factor = 2
+            trim = 0.12  # Trim 24% in x and y
+        elif radius <= 100:
+            shrink_factor = 4
+            trim = 0.16  # Trim 32% in x and y
+        else:
+            shrink_factor = 8
+            trim = 0.20  # Trim 40% in x and y
+
+        # Trim the ball kernel
+        n = int(ball.shape[0] * trim)
+        i0, i1 = n, ball.shape[0] - n
+        ball = ball[i0:i1, i0:i1]
+
+    # Rescale the image and ball kernel
+    image_rescaled = skimage.transform.rescale(
+        image, 1.0 / shrink_factor, preserve_range=True
+    ).astype(image.dtype)
+    kernel_rescaled = skimage.transform.rescale(
+        ball, 1.0 / shrink_factor, preserve_range=True
+    ).astype(ball.dtype)
+
+    # Compute the rolling ball background
+    background = skimage.restoration.rolling_ball(
+        image_rescaled, kernel=kernel_rescaled, **kwargs
+    )
+
+    # Apply Gaussian smoothing if specified
+    if smooth is not None:
+        background = skimage.filters.gaussian(
+            background, sigma=smooth / shrink_factor, preserve_range=True
+        )
+
+    # Resize the background to the original image size
+    background_resized = skimage.transform.resize(
+        background, image.shape, preserve_range=True
+    ).astype(image.dtype)
+
+    return background_resized
+
+
+def subtract_background(
+    image, radius=100, ball=None, shrink_factor=None, smooth=None, **kwargs
+):
+    """Subtract the background from an image using the rolling ball algorithm.
+
+    Args:
+        image (np.ndarray): Input image from which to subtract the background.
+        radius (int, optional): Radius of the rolling ball. Defaults to 100.
+        ball (np.ndarray, optional): Precomputed ball kernel. If None, it will be generated.
+        shrink_factor (int, optional): Factor to shrink the image and ball for faster computation.
+            Determined based on radius if not provided.
+        smooth (float, optional): Sigma for Gaussian smoothing applied to the background after
+            rolling ball. Defaults to None.
+        **kwargs: Additional arguments passed to the rolling_ball_background_skimage function.
+
+    Returns:
+        np.ndarray: Image with the background subtracted.
+    """
+    # Calculate the background using the rolling ball algorithm
+    background = rolling_ball_background_skimage(
+        image,
+        radius=radius,
+        ball=ball,
+        shrink_factor=shrink_factor,
+        smooth=smooth,
+        **kwargs,
+    )
+
+    # Ensure that the background does not exceed the image values
+    mask = background > image
+    background[mask] = image[mask]
+
+    # Subtract the background from the image
+    return image - background
+
+
+def apply_ic_field(
+    data,
+    correction=None,
+    zproject=False,
+    rolling_ball=False,
```

Comment:
Same goes for these:

https://github.com/CellProfiler/CellProfiler/blob/fa81fb0f2850c7c6d9cefdf4e71806188f1dc546/src/frontend/cellprofiler/modules/correctilluminationapply.py#L62

Should we review if these are equivalent? This is hard to deal with, and is painful because we can't just add CellProfiler as an imported package, so not sure what is best